src/Mentat.Repository/Services/CardService.cs

```diff
@@ -35,9 +30,38 @@ public void RemoveCard(string id)
             _cards.DeleteOne(card => card.Id.Equals(id));
         }
 
-        public void UpdateCard(string id, Card card)
+        public void SaveCard(string id, IFormCollection collection)
+        {
+            var card = new Card();
+            if (id != null)
+            {
+                card = GetCard(id);
+                if (card == null)
+                {
+                    throw new Exception("Card ID invalid.");
+                }
+            }
+
+            card.Notes = collection["notes"];
+            card.Subject = collection["subject"];
+            card.Question = collection["question"];
+            card.Answer = collection["answer"];
+            card.DifficultyLevel = collection["difficultyLevel"];
+
+            InsertOrUpdateCard(card);
+        }
+
+        private void InsertOrUpdateCard(Card card)
         {
-            _cards.ReplaceOne(card => card.Id.Equals(id), card);
+            if (card.Id == null)
+            {
+                card.Id = Guid.NewGuid().ToString();
+                _cards.InsertOne(card);
+            }
+            else
+            {
+                _cards.ReplaceOne(c => c.Id.Equals(card.Id), card);
+            }
         }
     }
 }
```

Comment:
Since this file has a lot of logic, can we get some comprehensive unti tests on this file? Here is an example of what I mean:
https://github.com/dyohn/mentat/blob/development/tests/Mentat.Domain.Tests/BashTestDriver.Tests.cs

Reply:
Hi, @jonnywwright,
I fought with this for a few hours trying to get a workable test and did not succeed. I do have other things I need to wrap up for this semester's class, and I am not confident that I can do both that and this, nor am I confident that I could do this alone. This is my first dealings with Mongo and considering that there are no current examples in our code base of unit tests dealing with Mongo and/or mocking Mongo, there's not a lot to go off of. 
Since my change to this file is -1 method, I feel like me not implementing test(s) should not stop this change.
Please let me know your comments and/or questions.
Rachel